frontend/api/views.py

```diff
@@ -0,0 +1,88 @@
+from rest_framework import serializers, status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from django.shortcuts import render
+from api.models import Consumer, MonthlyStatistics
+
+
+class ConsumerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Consumer
+        fields = '__all__'
+
+
+class MonthlyStatisticsSerializer(serializers.ModelSerializer):
+    consumer = ConsumerSerializer(read_only=True)
+
+    class Meta:
+        model = MonthlyStatistics
+        fields = '__all__'
+
+
+class ConsumerTypes(APIView):
+    def get(self, request):
+        return Response(Consumer.CONSUMER_TYPE_MAP)
+
+
+class ConsumerList(APIView):
+
+    def get(self, request, consumer_type=None):
+        filters = {}
+        if consumer_type:
+            filters['consumer_type'] = consumer_type
+
+        consumers = Consumer.objects.filter(**filters)
+
+        serializer = ConsumerSerializer(consumers, many=True)
+
+        return Response(serializer.data)
+
+
+class ConsumerDetail(APIView):
+
+    def get(self, request, consumer_id):
+        consumer = Consumer.objects.get(pk=consumer_id)
+        serializer = ConsumerSerializer(consumer)
+
+        return Response(serializer.data)
+
+    def post(self, request, consumer_id=None):
+        name = request.POST.get('name')
+        consumer_type = request.POST.get('consumer_type')
+        print(name, consumer_type)
+
+        consumer = Consumer.objects.create(name=name, consumer_type=consumer_type)
+
+        if consumer:
+            return Response(dict(success=True))
+
+        return Response(dict(sucess=False))
+
+    def delete(self, request, consumer_id):
+        consumer = Consumer.objects.get(pk=consumer_id)
+
+        if consumer:
+            consumer.delete()
+            return Response(dict(success=True))
+        return Response(dict(success=False))
```

Comment:
FYI: I used to use DRF viewsets for making view's functions.
http://www.django-rest-framework.org/api-guide/viewsets/

e.g.
```py
class UserViewSet(viewsets.ModelViewSet):
    """
    A viewset for viewing and editing user instances.
    """
    serializer_class = UserSerializer
    queryset = User.objects.all()
```

But, it's also good to evaluate what views functions applicants will make. 🙂

Reply:
I do not expect applicants touching backend.